src/mkdocstrings/handlers/python.py

```diff
@@ -290,3 +294,30 @@ def rebuild_category_lists(obj: dict) -> None:
     obj["children"] = [child for _, child in obj["children"].items()]
     for child in obj["children"]:
         rebuild_category_lists(child)
+
+
+def sort_object(obj: CollectorItem, sort_style: str) -> None:
+    """Sort the collected object's children.
+
+    Sorts the object's children list, then each category separately, and then recurses into each.
+
+    Arguments:
+        obj: The collected object, as a dict. Note that this argument is mutated.
+        sort_style: How to sort the children lists - 'lexical' or 'source'.
+    """
+
+    def sort_key(a: dict) -> Any:
+        if sort_style == "lexical":
+            return a.get("name")
+        elif sort_style == "source":
+            return a.get("source", {}).get("line_start", 0)
+        else:
+            raise Exception("unknown sort_style")
```

Comment:
Rather than defining a function containing the condition, I think it would be preferable to run the condition once to define the proper sorting function. For example:

```python
import operator

if sort_style == "alphabetical":
    sort_function = operator.itemgetter("name")
elif sort_style == "source":
    sort_function = lambda item: item.get("source", {}).get("line_start", 0)
```

Maybe we could even write those once and for all as private functions, and pass one or the other directly to the `sort_object` function based on the config.